### PR TITLE
Report KSP logs in compilations

### LINKED ISSUE #69: KSP errors are not included in the message output
I'm trying to use errors printed by KSP in my test but it seems that any KSP logs are not included. I see them in the test logs but not in `KotlinCompilation.Result.messages`.

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #69. This is based on an implementation I've been using in my projects, and also works around internal access across modules via lesser-known suppression mechanism until kotlin finally finishes implementing friend-paths support in their gradle/IDE tooling.

### CODE REVIEW

File: ksp/src/main/kotlin/com/tschuchort/compiletesting/MessageCollectorBasedKSPLogger.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.jetbrains.kotlin.cli.common.messages.CompilerMessageSeverity
+import org.jetbrains.kotlin.cli.common.messages.MessageCollector
+import com.google.devtools.ksp.processing.KSPLogger
+import com.google.devtools.ksp.symbol.FileLocation
+import com.google.devtools.ksp.symbol.KSNode
+import com.google.devtools.ksp.symbol.NonExistLocation
+import java.io.PrintWriter
+import java.io.StringWriter
+
+internal class MessageCollectorBasedKSPLogger(private val messageCollector: MessageCollector) : KSPLogger {
```
Comment (ZacSweers) [inline ksp/src/main/kotlin/com/tschuchort/compiletesting/MessageCollectorBasedKSPLogger.kt:12]: Note this implementation is based on the one in KSP itself

**Comment by tschuchortdev** [comment]
Do we need a tests for this, to make sure something like #69 doesn't happen again?

**Comment by ZacSweers** [comment]
Actually, this doesn't resolve #69 directly (doesn't cover errors). Just a step toward it. My linked workaround is still an issue that I need to work around this. Let me push a proposal idea

**Comment by ZacSweers** [comment]
Actually this should work, let me confirm

File: ksp/src/main/kotlin/com/tschuchort/compiletesting/MessageCollectorBasedKSPLogger.kt (hunk 1 above)
Comment (ZacSweers) [reply ksp/src/main/kotlin/com/tschuchort/compiletesting/MessageCollectorBasedKSPLogger.kt]: Removed, I got my wires crossed

**Comment by ZacSweers** [comment]
Confirmed this worked and added extra tests for it 👍

**Pull request merged**